### ISSUE TITLE
Add option to put build products in Xcode's DerivedData

### ISSUE THREAD

**Issue by tkrajacic** [realm/jazzy#302]
At the moment _xcodebuild_ creates a `build/` dir for output in the projects directory.

One could set the `CONFIGURATION_BUILD_DIR` environment variable but it would be great if _jazzy_ would automatically use the `DerivedData/` directory from the project file.

The directory for the project could probably be obtained via:

`xcodebuild -project MyProject.xcodeproj -showBuildSettings`

Sure, one could just `.gitignore` the `build/` directory (which everyone hopefully does anyway) but it is just one more directory uselessly cluttering the project.

**Comment by jpsim** [comment]
I absolutely understand why you'd want to avoid writing to a `build/` directory, and I'd definitely recommend setting `CONFIGURATION_BUILD_DIR` to do this.

It's not apparent to me that adding a new configuration parameter would be worthwhile given that it does the same thing as an existing environment variable, only with a different default, and would go against what users may already be familiar with `xcodebuild`.

**Comment by tkrajacic** [comment]
I agree it was a bad idea suggesting a new configuration parameter.

The proposal should have been:
_Jazzy should honor the build directory of the project_

Would that make sense to you?

**Comment by jpsim** [comment]
Jazzy shells out to `xcodebuild`, and if a custom build directory is set in the Xcode project, it should respect that. However, `Xcode.app` and `xcodebuild` have different *defaults*. It may be preferable for jazzy to override the default to mimic `Xcode.app` but I'm still not convinced.

**Comment by tkrajacic** [comment]
Maybe you could leave the issue open? See what others think.

I am still strongly favoring jazzy using the DerivedData directory as default, as it would be difficult to manually set this (you can't know the exact directory name including the hash value unless you check the project file manually)

**Comment by pcantrell** [comment]
I’m sure I’d have spent an hour digging for the output the first time I used Jazzy if it defaulted to DerivedData.

**Comment by tkrajacic** [comment]
@pcantrell I think you are mistaking 'output' in this context. I don't mean the documentation! I mean the build products of `xcodebuild` which you most certainly never need...

**Comment by pcantrell** [comment]
Ah, I see what you mean. Using DerviedData (or any tmp dir) does make sense.

But why don't I get a `build` dir when I run jazzy? My project doesn't override `CONFIGURATION_BUILD_DIR`. I do override the derived data dir globally in my Xcode prefs … but isn't that what you're saying doesn't work and you want to work?

**Comment by tkrajacic** [comment]
I think you are seeing what @jpsim was mentioning (and I didn't know).
When you do not set a build directory implicitly, _Xcode_ and _xcodebuild_ have different defaults.

_Xcode_ defaults to `DerivedData/`, and _xcodebuild_ to a local `build/` directory.

When you __do__ set a directory then _jazzy_ will pick up on it.
Since I (and I can only assume most others?) am using the default `DerivedData/` _jazzy_ doesn't see anything set and _xcodebuild_ will go ahead with its own default of `build/`.

I am voting for _jazzy_ using _Xcode's_ default of `DerivedData/` here if nothing is specified within _Xcode_.

**Comment by tkrajacic** [comment]
@jpsim Would you mind leaving this open (re-open it), or is anything like discussed here completely out of the question? I don't think the request is unreasonable, or maybe someone will come up with a better solution.

**Comment by jpsim** [comment]
You're absolutely right that the request is reasonable, I just can't think of a solution that wouldn't be just as confusing to others. If anyone can make a PR that does this in a nice way, without shifting the confusion from one set of users to another, I'd be happy to consider it.

**Comment by istx25** [comment]
Any updates on this?

**Comment by jpsim** [comment]
No progress to report.

**Comment by drekka** [comment]
I came across this trying to figure out how to avoid creating the 'build' directory in my project every time I run Jazzy. I'm using defaults in my project.

I tried passing a 'BUILD_DIR', 'CONFIGURATION_BUILD_DIR' and several other settings, but Jazzy still created the build dir and stored. I also tried passing '-x derivedDataPath xxoooxx' with no luck either.

All I want to do is to change from creating a directory called 'build' to a temporary directory I'll create manually.

**Comment by johnfairh** [comment]
I think you want `-x SYMROOT=wherever`

**Comment by drekka** [comment]
Hi. I tried using the SYMROOT option but it didn't work. I ended up figuring out how to use the `-x` option to set the derivedDataPath. It wouldn't work until I added the `-scheme` option as well, but it's now working fine and putting the build files in the temporary directory I pass to it.

     jazzy --xcodebuild-arguments "-configuration,$CONFIGURATION,-derivedDataPath,$TMP_DIR,-scheme,ReleaseScheme" ...

**Comment by SDGGiesbrecht** [comment]
Hurray! Thanks for the tip @drekka.

I fought with this a while back and ended up giving up. Now I will take another look.
